frontend/src/concepts/k8s/NameDescriptionField.tsx

```diff
@@ -42,14 +50,20 @@ const NameDescriptionField: React.FC<NameDescriptionFieldProps> = ({
   return (
     <Stack hasGutter>
       <StackItem>
-        <FormGroup label="Name" isRequired fieldId={nameFieldId}>
+        <FormGroup
+          label="Name"
+          isRequired
+          fieldId={nameFieldId}
+          helperText={isNameValid && nameHelperText}
+        >
           <TextInput
             isRequired
             ref={autoSelectNameRef}
             id={nameFieldId}
             name={nameFieldId}
             value={data.name}
             onChange={(name) => setData({ ...data, name })}
+            validated={!isValidNotebookName(data.name) ? 'error' : undefined}
```

Comment:
`NameDescriptionField` should not call `isValidNotebookName` directly. It should either be a validation function passed in or use the newly added `isNameValid` value to determine whether or not to display the error.